### PR TITLE
Adding sanity test for zero copy io

### PR DESCRIPTION
**Pull request by AmarnatReddy**
# Description
Adding sanity test for zero copy io

This test is being added in reference to the design document:
https://docs.google.com/document/d/1iabmAgul2mpSghIvjot5gMBQRFMrS1rbWjtc8HNM-8Q/edit?tab=t.0#heading=h.vmyuqctl6kpk

It aims to cover both basic acceptance and functional validations for zero-copy IO workflows.

Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-BYCCZ6

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_mount_data_consistency_tests.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    )
+
+    # Read and calculate checksum from reader
+    checksum_reader, _ = reader.exec_command(
+        sudo=True, cmd=f"sha256sum {file_path_reader} | awk '{{print $1}}'"
+    )
+
+    if checksum_writer.strip() != checksum_reader.strip():
+        raise CommandFailed(
+            f"Checksum mismatch: Writer ({file_path_writer}) vs Reader ({file_path_reader})"
```
Comment (Manimaran-MM) [inline tests/cephfs/cephfs_mount_data_consistency_tests.py:43]: Can we add the debug logs for the values that are getting compared?
Something similar to 
raise CommandFailed(
            "Checksum mismatch: Writer ({}) vs Reader ({}). Checksum Writer: {} Checksum Reader: {}".format(file_path_writer, file_path_reader, checksum_writer.strip(),  checksum_reader.strip())

File: tests/cephfs/cephfs_mount_data_consistency_tests.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        )
+
+
+def write_and_verify_data(client, file_path, data):
+    client.exec_command(sudo=True, cmd=f"echo '{data}' > {file_path}")
+    out, _ = client.exec_command(sudo=True, cmd=f"cat {file_path}")
+    if out.strip() != data:
+        raise CommandFailed(f"Data mismatch at {file_path}")
```
Comment (Manimaran-MM) [inline tests/cephfs/cephfs_mount_data_consistency_tests.py:51]: Same debug logs as above

**Review by neha-gangadhar — APPROVED** [review]
Approved
